scripts/o3de/o3de/compatibility.py

```diff
@@ -185,21 +185,38 @@ def get_incompatible_gem_version_specifiers(gem_version_specifier_list:list, all
         return set(gem_version_specifier_list)
 
     incompatible_gem_version_specifiers = set()
+
+    # helper function to check dependency tree for incompatible gem dependencies
+    def get_gem_dependency_version_specifiers(gem_name):
+        gem_dependencies = all_gems_json_data[gem_name].get('dependencies')
+        if gem_dependencies:
+            incompatible_dependency_specifiers = get_incompatible_gem_version_specifiers(gem_dependencies, all_gems_json_data, checked_specifiers)
+            if incompatible_dependency_specifiers:
+                incompatible_gem_version_specifiers.update(incompatible_dependency_specifiers)
+
     for gem_version_specifier in gem_version_specifier_list:
+        if gem_version_specifier in checked_specifiers:
+            continue
+
+        checked_specifiers.add(gem_version_specifier)
+
         gem_name, version_specifier = utils.get_object_name_and_optional_version_specifier(gem_version_specifier)
         if not gem_name in all_gems_json_data:
             incompatible_gem_version_specifiers.add(f"{gem_version_specifier} (missing dependency)")
             continue
```

Comment:
What is the error message around if a dependent gem is missing an dependency.

Is it is something such as "Atom could be activated as one of its dependencies("Atom_Feature_Common") has a missing dependency on "ImGui"?

Reply:
Hmmm, it just shows the missing dependency - but it doesn't state what gem it is a dependency of - e.g.
Given:
`Gem A` which depends on `Gem B` which depends on `Gem C`
When:
`Gem C` is missing, you just see the error that `Gem C` is missing, but no that it is a dependency of `Gem B`

Reply:
I've updated the output to include gem names. e.g.
GemA is missing the depencency GemB>1.2.3
GemB depends on GemC==2.3.4 but GemC version 3.2.1 was found